### PR TITLE
TestFrameworkConfigPathProvider needs strings for all inputs, not nulls

### LINKED ISSUE #288: Still can't configure Infection
| Question    | Answer
| ------------| ---------------
| Infection version | master

```
Any directories to exclude from within your source directories?: 
Where is your .(xml|yml)(.dist) configuration located?: 
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Infection\Config\ValueProvider\TestFrameworkConfigPathProvider::Infection\Config\ValueProvider\{closure}() must be of the type string, null given in src/Config/ValueProvider/TestFrameworkConfigPathProvider.php:79
Stack trace:
#0 [internal function]: Infection\Config\ValueProvider\TestFrameworkConfigPathProvider->Infection\Config\ValueProvider\{closure}(NULL)
#1 vendor/symfony/console/Helper/QuestionHelper.php(409): call_user_func(Object(Closure), NULL)
#2 vendor/symfony/console/Helper/QuestionHelper.php(65): Symfony\Component\Console\Helper\QuestionHelper->validateAttempts(Object(Closure), Object(Symfony\Component\Console\Output\StreamOutput), Object(Symfony\Component\Console\Question\Question))
#3 src/Config/ValueProvider/TestFrameworkConfigPathProvider.php(108): Symfony\Component\Console\Helper\QuestionHelper->ask(Object(Symfony\Component\Co in src/Config/ValueProvider/TestFrameworkConfigPathProvider.php on line 79
```

### PR DESCRIPTION
**Pull request by sanmai**
- [x] Fixes #288
- [x] Adds an E2E test for initial configuration

### CODE REVIEW

File: src/Config/ValueProvider/TestFrameworkConfigPathProvider.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ private function getValidator(string $testFramework): \Closure
         };
     }
 
-    private function askTestFrameworkConfigLocation(InputInterface $input, OutputInterface $output, array $dirsInCurrentDir, string $testFramework, $defaultValue): string
+    private function askTestFrameworkConfigLocation(InputInterface $input, OutputInterface $output, array $dirsInCurrentDir, string $testFramework, string $defaultValue): string
```
Comment (sidz) [inline src/Config/ValueProvider/TestFrameworkConfigPathProvider.php]: what about to add default value ?

Comment (sanmai) [reply src/Config/ValueProvider/TestFrameworkConfigPathProvider.php]: It is set explicitly on lines 53 and 57 above

File: tests/Fixtures/e2e/Configure/do_configure.expect [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+configure "directories do you want to include"
+configure "Any directories to exclude from"
+configure "configuration located?"
+configure "timeout in seconds"
+configure "text log file?"
+
+expect "Please note that some mutants will inevitably be harmless"
+#expect eof
```
[the remark this answers is not in the document]
Comment (sanmai) [reply tests/Fixtures/e2e/Configure/do_configure.expect]: OK, done.

**Comment by borNfreee** [comment]
Thank you @sanmai 👍

**Pull request merged**